staging/src/k8s.io/noderesourcetopology-api/pkg/apis/topology/v1alpha1/types.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	intstr "k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourceTopology is a specification for a Foo resource
+type NodeResourceTopology struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	TopologyPolicies []string `json:"topologyPolicies"`
+	Zones            ZoneMap  `json:"zones"`
+}
+
+// Zone is the spec for a NodeResourceTopology resource
+type Zone struct {
+	Type       string          `json:"type"`
+	Parent     string          `json:"parent,omitempty"`
+	Costs      map[string]int  `json:"costs,omitempty"`
+	Attributes map[string]int  `json:"attributes,omitempty"`
+	Resources  ResourceInfoMap `json:"resources,omitempty"`
+}
+
+type ZoneMap map[string]Zone
```

Comment:
We almost never do maps in k8s APIs.  I think we want this to be a list of structs with a "name" key.

Reply:
is it a strict rule? or you are afraid generator/verifier could fail. In this case generator managed to produce working code.

Reply:
ok, no problem, initial version of this API included list of structures.

Reply:
We moved away from maps to lists of structures.